ui/components/app/transaction-activity-empty-state/transaction-activity-empty-state.test.tsx

```diff
@@ -67,4 +103,106 @@ describe('TransactionActivityEmptyState', () => {
       './images/empty-state-activity-dark.png',
     );
   });
+
+  describe('Swap button functionality', () => {
+    const accountWithoutSigning: InternalAccount = {
+      ...mockAccount,
+      methods: ['personal_sign'], // No eth_signTransaction or eth_signUserOperation
+    };
+
+    const accountWithSigning: InternalAccount = {
+      ...mockAccount,
+      methods: [EthMethod.SignTransaction, 'personal_sign'],
+    };
+
+    it('disables swap button when not a swaps chain', () => {
+      // Override state to simulate testnet (non-swaps chain)
+      const testnetState = {
+        metamask: {
+          ...mockState.metamask,
+          useExternalServices: true,
+          selectedNetworkClientId: 'goerli',
+          networkConfigurationsByChainId: {
+            ...mockState.metamask.networkConfigurationsByChainId,
+            '0x5': {
+              // Goerli testnet - not in allowed swaps chains
+              chainId: '0x5',
+              name: 'Goerli',
+              nativeCurrency: 'ETH',
+              defaultRpcEndpointIndex: 0,
+              rpcEndpoints: [
+                {
+                  type: 'infura',
+                  url: 'https://goerli.infura.io/v3/test',
+                  networkClientId: 'goerli',
+                },
+              ],
+              blockExplorerUrls: [],
+            },
+          },
+        },
+      };
+
+      renderComponent({ account: accountWithSigning }, testnetState);
+      const swapButton = screen.getByRole('button', { name: 'Swap tokens' });
+      expect(swapButton).toBeDisabled();
+    });
+
+    it('disables swap button when external services are disabled', () => {
+      const stateWithoutExternalServices = {
+        metamask: {
+          ...mockState.metamask,
+          useExternalServices: false,
+        },
+      };
+
+      renderComponent(
+        { account: accountWithSigning },
+        stateWithoutExternalServices,
+      );
+      const swapButton = screen.getByRole('button', { name: 'Swap tokens' });
+      expect(swapButton).toBeDisabled();
+    });
+
+    it('disables swap button when account cannot sign transactions', () => {
+      renderComponent({ account: accountWithoutSigning });
+      const swapButton = screen.getByRole('button', { name: 'Swap tokens' });
+      expect(swapButton).toBeDisabled();
+    });
```

Comment:
There are three conditionals that should trigger the disabled state of the swap button `!isSwapsChain || !isSigningEnabled || !isExternalServicesEnabled,`